galsim/compound.py

```diff
@@ -902,3 +904,162 @@ def __setstate__(self, d):
+    @param  hlr      Half light radius of the distribution of points.  This
+                     is the mean half light radius produced by an 
+                     infinite number of points.  A single instance will be
+                     noisy.
+    @param  flux     Total flux in all point sources.  Default 1
+    @param  nstep    Number of steps in random walk.  Default 40
+    @param  rng      Optional random number generator. Should be a UniformDeviate.
+    @param  gsparams GSParams for the gaussians representing each point source.
+
+    Methods
+    -------
+
+    This class inherits from galsim.Sum. Additional methods, implemented as
+    read-only properties, are provided as "getters" for the basic parameters:
+
+        .npoints
+        .half_light_radius
+        .flux
+        .nstep
+        .gaussians
+            The list of galsim.Gaussian objects representing the points
+
+    Notes
+    -----
+
+    - The algorithm is a modified version of that presented in
+      https://arxiv.org/abs/1312.5514v3 : There is no outer cutoff to how far a
+      point can wander.
+    - This class inherits from galsim.Sum
+    """
+    def __init__(self, npoints, hlr, flux=1.0, nstep=40, rng=None, gsparams=None):
+
+        self._hlr     = float(hlr)
+        self._flux    = float(flux)
+        self._npoints = int(npoints)
+        self._nstep   = int(nstep)
+
+        self._gaussian_sigma = 1.0e-8
+
+        self._input_gsparams=gsparams
+
+        if rng is None:
+            rng = galsim.UniformDeviate()
+
+        self._rng=rng
+
+        self._verify()
+
+        # this is the scale factor by which to multiply each step
+        # in order to get the requested half light radius
+        factor = numpy.sqrt(nstep)/2.09
+        self._scale = hlr/factor
+
+        pts = self._get_points()
+        self._gaussians = self._get_gaussians(pts)
+
+        super(RandomWalk, self).__init__(self._gaussians)
+
+    @property
+    def half_light_radius(self):
+        return self._hlr
+
+    @property
+    def flux(self):
+        return self._flux
+
+    @property
+    def npoints(self):
+        return self._npoints
+
+    @property
+    def nstep(self):
+        return self._nstep
+
+    @property
+    def gaussians(self):
+        return self._gaussians
+
+
+    def _get_gaussians(self, points):
+
+        fluxper=self._flux/self._npoints
+        gaussians=[]
+
+        for i in xrange(points.shape[0]):
+            dx,dy = points[i]
+
+            g=galsim.Gaussian(
+                sigma=self._gaussian_sigma,
+                flux=fluxper,
+                gsparams=self._input_gsparams,
+            )
+            g = g.shift(dx=dx, dy=dy)
+
+            gaussians.append(g)
+
+        return gaussians
+
+    def _get_points(self):
+
+        scale=self._scale
+        npoints=self._npoints
+        nstep=self._nstep
+
+        pts=numpy.zeros( (npoints, 2) )
+
+        rng=self._rng
+
+        for i in xrange(npoints):
```

Comment:
For Python 3 compatibility, this should be simply `range`.  If you think the list construction overhead is significant, you can check [here](http://python-future.org/compatible_idioms.html) for the python 2/3 compatible construction, but I suspect it's fine to just use `range`.

Update: the 2/3 compatible thing is to just put `from builtins import range` at the top, so that's pretty easy.